src/security_utilities_rust/src/microsoft_security_utilities_core/cross_company_correlating_id.rs

```diff
@@ -13,23 +13,25 @@ thread_local! {
 pub fn generate_cross_company_correlating_id(text: &str) -> String {
     let hash = generate_sha256_hash(text);
 
-    let hash = format!("CrossMicrosoftCorrelatingId:{}", hash);
-
     let checksum = THREAD_LOCAL_SHA256.with(|sha| {
-        sha.borrow_mut().update(hash.as_bytes());
-        sha.borrow_mut().finalize_reset()
+        let mut sha = sha.borrow_mut();
+
+        sha.update("CrossMicrosoftCorrelatingId:");
+        sha.update(hash);
+        sha.finalize_reset()
     });
 
     let to_encode = &checksum[0..15];
     STANDARD.encode(to_encode)
 }
 
-pub fn generate_sha256_hash(text: &str) -> String {
-
+fn generate_sha256_hash(text: &str) -> String {
     let result = THREAD_LOCAL_SHA256.with(|sha| {
         sha.borrow_mut().update(text.as_bytes());
         sha.borrow_mut().finalize_reset()
     });
 
-    fmt::format(format_args!("{:X}", result))
-}
\ No newline at end of file
+    let mut output = String::with_capacity(2 * 32);
```

Comment:
We know this value exactly because hex encoding `n` bytes will always produce `n * 2` ascii bytes. 

Now that you point it out, I actually realised we don't need this allocation either: 
1. We know all the size up front.
2. The encoding is super simple and easy to verify.

Will add a commit that removes this one too, since we're hunting for % anyways.

Reply:
Done, this should now be ready to merge :)